src/Type/MixedType.php

```diff
@@ -403,6 +403,18 @@ public function toBoolean(): BooleanType
 
 	public function toNumber(): Type
 	{
+		if ($this->subtractedType !== null) {
+			$float = new FloatType();
+			$int = new IntegerType();
+
+			if ($float->isSuperTypeOf($this->subtractedType)->yes()) {
+				return $int;
+			}
+			if ($int->isSuperTypeOf($this->subtractedType)->yes()) {
+				return $float;
+			}
```

Comment:
What if both `float` and `int` are subtracted?

Reply:
sorry, can someone explain to me how this works in general? if something is e.g. not a float it could be e.g. an object still and objects can't be incremented?

Reply:
Even if float is subtracted, the result can still be a float... Because `MixedType` without `FloatType` can still be numeric-string for example... https://3v4l.org/StqSP